maas/resource_maas_network_interface_physical.go

```diff
@@ -34,18 +34,8 @@ func resourceMaasNetworkInterfacePhysical() *schema.Resource {
 				if err != nil {
 					return nil, err
 				}
-				tfState := map[string]interface{}{
-					"id":          fmt.Sprintf("%v", n.ID),
-					"mac_address": n.MACAddress,
-					"machine":     machine.SystemID,
-					"mtu":         n.EffectiveMTU,
-					"name":        n.Name,
-					"tags":        n.Tags,
-					"vlan":        n.VLAN.ID,
-				}
-				if err := setTerraformState(d, tfState); err != nil {
-					return nil, err
-				}
+				d.Set("machine", idParts[0])
+				d.SetId(fmt.Sprintf("%v", n.ID))
```

Comment:
Considering that [ID](https://github.com/maas/gomaasclient/blob/78ffc951871382b09ea39a96eed24964df038b46/entity/network_interface.go#L25) is an `int` we can use https://pkg.go.dev/strconv#Itoa

More on this topic https://dev.to/mavensingh/strconvitoa-vs-strconvformatint-vs-fmtsprintf-which-is-good-to-convert-int-to-string-in-golang-3iol

Reply:
Thank you for the suggestion. I also applied this to all other usages of Sprintf in this file